src/SickSafetyscanners.cpp

```diff
@@ -245,7 +245,13 @@ void SickSafetyscanners::startTCPConnection(const sick::datastructure::CommSetti
       boost::ref(*m_io_service_ptr),
       settings.getSensorIp(),
       settings.getSensorTcpPort());
-  async_tcp_client->doConnect();
+
+  if (!async_tcp_client->doConnect())
```

Comment:
The default `wait_for_connect` value is -1 seconds, which means "block forever until connected", correct?
Do you want to put a non-zero wait time here?